miner/worker_builder.go

```diff
@@ -451,9 +454,10 @@ func (w *worker) simulateBundle(
 			// for unRevertible tx but itself can be dropped, we drop it and revert the state and gas pool
 			if containsHash(bundle.DroppingTxHashes, receipt.TxHash) {
 				log.Warn("drop tx in bundle", "hash", receipt.TxHash.String())
-				state = prevState
-				gasPool = prevGasPool
+				// do not need to revert the state and gas pool for they are already reverted in ApplyTransaction
```

Comment:
"// do not need to revert the state and gas pool for they are already reverted in ApplyTransaction" I did not find that

Reply:
here https://github.com/bnb-chain/bsc-builder/blob/main/core/vm/evm.go#L277